framework/src/base/FEProblem.C

```diff
@@ -1202,22 +1194,14 @@ FEProblem::addFunction(std::string type, const std::string & name, InputParamete
   for (THREAD_ID tid = 0; tid < libMesh::n_threads(); tid++)
   {
     MooseSharedPointer<Function> func = MooseSharedNamespace::static_pointer_cast<Function>(_factory.create(type, name, parameters, tid));
-    if (_functions[tid].find(name) != _functions[tid].end())
-      mooseError("Duplicate function name added to FEProblem: " << name);
```

Comment:
impossible eh? 
Challenge excepted!
Thanks for the code reference, I'm convinced enough for the purpose of this PR.